apps/hash-graph/libs/api/src/rest/data_type.rs

```diff
@@ -203,31 +205,25 @@ where
     let is_list = matches!(&schema, ListOrValue::List(_));
 
     let schema_iter = schema.into_iter();
-    let mut data_types = Vec::with_capacity(schema_iter.size_hint().0);
-    let mut partial_metadata = Vec::with_capacity(schema_iter.size_hint().0);
-
-    for data_type in schema_iter {
-        domain_validator.validate(&data_type).map_err(|report| {
-            tracing::error!(error=?report, id=data_type.id().to_string(), "Data Type ID failed to validate");
-            StatusCode::UNPROCESSABLE_ENTITY
-        })?;
-
-        partial_metadata.push(PartialDataTypeMetadata {
-            record_id: data_type.id().clone().into(),
-            classification: OntologyTypeClassificationMetadata::Owned { owned_by_id },
-        });
-
-        data_types.push(data_type);
-    }
 
     let mut metadata = store
         .create_data_types(
             actor_id,
             &mut authorization_api,
             temporal_client.as_deref(),
-            data_types.into_iter().zip(partial_metadata),
-            ConflictBehavior::Fail,
-            relationships,
+            schema_iter.into_iter().map(|schema| {
+                domain_validator.validate(&schema).map_err(|report| {
+                    tracing::error!(error=?report, id=schema.id().to_string(), "Data Type ID failed to validate");
+                    StatusCode::UNPROCESSABLE_ENTITY
```

Comment:
down there changed to `hash_status`, maybe here as well?

Reply:
The return value of the Function is of type `http::StatusCode`, thus this must not be a `hash_status`. As we're going to switch to our own RPC implementation soon(-ish) I'm only going to change the signatures which are required in this PR.